twake/backend/node/src/services/channels/services/cassandra.ts

```diff
@@ -1,35 +1,82 @@
-/* eslint-disable @typescript-eslint/no-unused-vars */
 import cassandra from "cassandra-driver";
 import { Channel } from "../entities";
-import ChannelServiceAPI from "../provider";
+import ChannelServiceAPI, { ChannelPrimaryKey } from "../provider";
 import {
-  UpdateResult,
-  CreateResult,
   DeleteResult,
+  OperationType,
+  SaveResult,
 } from "../../../core/platform/framework/api/crud-service";
+import { WorkspaceExecutionContext } from "../types";
+import { plainToClass } from "class-transformer";
 
 export class CassandraChannelService implements ChannelServiceAPI {
   version = "1";
+  keyspace = "twake";
+  table = "channels";
 
   constructor(private client: cassandra.Client) {}
 
-  create(channel: Channel): Promise<CreateResult<Channel>> {
-    throw new Error("Not implemented");
+  async save(channel: Channel): Promise<SaveResult<Channel>> {
+    const mode = channel.id ? OperationType.UPDATE : OperationType.CREATE;
+    const query = `INSERT INTO ${this.keyspace}.${this.table}
+      (
+      "company_id",
+      "workspace_id",
+      "id",
+      "icon",
+      "name",
+      "description",
+      "channel_group",
+      "visibility",
+      "is_default",
+      "archived"
+      )
```

Comment:
Can we map over parameters of the entity in the future here?

Reply:
Yes this is the idea. With some decorators it will be automatic and so we
will avoid writing queries.